src/ia-anniversary-banner.ts

```diff
@@ -0,0 +1,194 @@
+import { html, css, LitElement, property, customElement } from 'lit-element';
+import { nothing } from 'lit-html';
+import CloseCircleIcon from '@internetarchive/icon-close-circle';
+
+import { IAFile, IAMD, TimelineMoment } from './interfaces';
+
+import { formatMoments } from './utils';
+import wayforwardSvg from './images/way-forward-svg';
+import wayforwardSvgMin from './images/way-forward-min-svg';
+
+import './marquee';
+
+@customElement('ia-anniversary-banner')
+export class IaAnniversaryBanner extends LitElement {
+  @property({ type: Object }) iaFiles: IAFile[] = [];
+
+  @property({ type: Object }) iaMD: IAMD = { directory: '', separator_dir: '' };
+
+  @property({ type: Object }) moments: { [key: string]: TimelineMoment } = {};
+
+  @property({ type: Array }) shuffledMoments: TimelineMoment[] = [];
+
+  @property({ type: String }) landingURL = 'https://archive.org';
+
+  @property({ type: String }) viewMode = 'open'; // open | closed
+
+  @property({ type: Number }) hideBannerDays = 7;
+
+  @property({ type: String }) directory = '';
+
+  updated(changed: any) {
+    if (changed.has('iaFiles') || changed.has('iaMD')) {
+      this.shuffledMoments = [];
+      const { directory } = this.iaMD;
+      this.directory = directory;
+    }
+    if (changed.has('directory')) {
+      this.shuffleMoments(this.directory);
+    }
+  }
+
+  shuffleMoments(directory: string) {
+    const moments: TimelineMoment[] = formatMoments(
+      this.iaFiles,
+      this.iaMD,
+      this.landingURL,
+      directory
+    );
+
+    const newMoments = [...moments];
+    this.shuffledMoments = newMoments;
+
+    this.dispatchEvent(
+      new CustomEvent('momentsShuffled', {
+        detail: { moments: this.shuffledMoments },
+      })
+    );
+  }
+
+  private closeBanner() {
+    this.viewMode = 'closed';
+
+    if (this.hideBannerDays > 0) {
+      (window as any)?.Cookies.set('anniv-banner', 'x', {
```

Comment:
Can we use an installable cookie library like https://github.com/markcellus/cookie-store instead of relying on the one that only lives in petabox? This doesn't exist in offshoot.

Reply:
yes I thought of that, I really don't see this banner coming into offshoot as it will be just in Petabox homepage. do we need a second cooke function?

Reply:
The issue isn't so much about offshoot, but you've now made `window.Cookie` a dependency of your component, which will only exist in petabox. Using an installable package makes your component independent of the environment that it's in. 

Also, this cookie library is actually a polyfill for the new [Cookie Store API](https://developer.mozilla.org/en-US/docs/Web/API/Cookie_Store_API) that will eventually be native in browsers so it's not really a standalone cookie library.

Reply:
true ^_^

Reply:
will use a custom event to fire and let PB consumer set the cookie to keep this component clean.

Reply:
Ooh I like!